be/src/vec/runtime/vdata_stream_recvr.cpp

```diff
@@ -161,9 +162,11 @@ void VDataStreamRecvr::SenderQueue::add_block(const PBlock& pblock, int be_numbe
     COUNTER_UPDATE(_recvr->_deserialize_row_batch_timer, deserialize_time);
     COUNTER_UPDATE(_recvr->_decompress_timer, block->get_decompress_time());
     COUNTER_UPDATE(_recvr->_decompress_bytes, block->get_decompressed_bytes());
-
-    _block_queue.emplace_back(std::move(block), block_byte_size);
-    _update_block_queue_empty();
+    {
+        std::lock_guard<std::mutex> l(_lock);
+        _block_queue.emplace_back(std::move(block), block_byte_size);
+        _update_block_queue_empty_in_lock();
```

Comment:
Maybe we can wrap this piece of code into a function
```
lock
push block to queue
update emtpy flag
```

Reply:
that‘s right. will do.